pkg/api/agentPoolOnlyApi/vlabs/types.go

```diff
@@ -30,15 +30,37 @@ type ManagedCluster struct {
 // Properties represents the ACS cluster definition
 type Properties struct {
 	ProvisioningState       ProvisioningState        `json:"provisioningState,omitempty"`
-	KubernetesVersion       string                   `json:"kubernetesVersion"`
+	KubernetesVersion       string                   `json:"kubernetesVersion,omitempty"`
+	OrchestratorProfile     *OrchestratorProfile     `json:"orchestratorProfile,omitempty"`
 	DNSPrefix               string                   `json:"dnsPrefix" validate:"required"`
 	FQDN                    string                   `json:"fqdn,omitempty"`
 	AgentPoolProfiles       []*AgentPoolProfile      `json:"agentPoolProfiles,omitempty" validate:"dive,required"`
 	LinuxProfile            *LinuxProfile            `json:"linuxProfile,omitempty" validate:"required"`
 	WindowsProfile          *WindowsProfile          `json:"windowsProfile,omitempty"`
 	ServicePrincipalProfile *ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
 	AccessProfiles          map[string]AccessProfile `json:"accessProfiles,omitempty"`
-	CertificateProfile      *CertificateProfile      `json:"certificateProfile,omitempty" validate:"required"`
+	CertificateProfile      *CertificateProfile      `json:"certificateProfile,omitempty"`
+	AzProfile               *AzProfile               `json:"azProfile,omitempty"`
```

Comment:
If we're having second thoughts about AzProfile is it worth leaving it out for now?

Reply:
Then we need to figure out a different way for delivering azure.conf inside the VM. FWIW, I have opened https://github.com/kubernetes/kubernetes/issues/64912.

Reply:
Agreed, but given this api is shared with aks, I'm reluctant to add something we're not 100% sure about.  If we become sure we need it, we can add it subsequently.

Reply:
I can move it inside openShiftConfig. What's the alternative?

Reply:
I feel like in our conversations we are on the fence about using this structure to hold anything for OpenShift.  There has also been talk about a different top level resource (other than `ManageCluster`).  

Should we hold for now?